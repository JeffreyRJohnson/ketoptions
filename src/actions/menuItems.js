import uuid from 'uuid';
import database from '../firebase/firebase';

// ADD_MENU_ITEM
export const addMenuItem = menuItem => ({
  type: 'ADD_MENU_ITEM',
  menuItem
});

export const startAddMenuItem = (menuItemData = {}) => {
  return dispatch => {
    const {
      calories = 0,
      carbs = 0,
      fat = 0,
      menu_item = '',
      protein = 0,
      restaurant_name = ''
    } = menuItemData;

    const menuItem = {
      calories,
      carbs,
      fat,
      menu_item,
      protein,
      restaurant_name
    };

    return database
      .ref('menuItems')
      .push(menuItem)
      .then(ref => {
        dispatch(
          addMenuItem({
            id: ref.key,
            ...menuItem
          })
        );
      });
  };
};

// REMOVE_MENU_ITEM
export const removeMenuItem = ({ id } = {}) => ({
  type: 'REMOVE_MENU_ITEM',
  id
});

// EDIT_MENU_ITEM
export const editMenuItem = (id, updates) => ({
  type: 'EDIT_MENU_ITEM',
  id,
  updates
});
