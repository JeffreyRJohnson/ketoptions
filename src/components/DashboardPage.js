import React from 'react';
import MenuItemList from './MenuItemList';
import MenuItemListFilters from './MenuItemListFilters';

const DashboardPage = () => (
  <div>
    <MenuItemListFilters />
    <MenuItemList />
  </div>
);

export default DashboardPage;
