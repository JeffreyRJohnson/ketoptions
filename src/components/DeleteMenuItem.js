import React from 'react';

const DeleteMenuItem = () => <div>This is my delete Menu item component</div>;

export default DeleteMenuItem;
