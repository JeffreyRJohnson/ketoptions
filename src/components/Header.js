import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1 className="header">Ketoptions</h1>
    <NavLink
      to="/"
      activeClassName="is-active"
      exact={true}
      className="header__link"
    >
      Landing Page
    </NavLink>
    <NavLink
      to="/dashboard"
      activeClassName="is-active"
      className="header__link"
    >
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active" className="header__link">
      Create
    </NavLink>
    <NavLink to="/edit" activeClassName="is-active" className="header__link">
      Edit
    </NavLink>
    <NavLink to="/delete" activeClassName="is-active" className="header__link">
      Delete
    </NavLink>
  </header>
);

export default Header;
