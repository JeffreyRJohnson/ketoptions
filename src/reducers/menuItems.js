// MenuItems Reducer

const menuItemsReducerDefaultState = [];

export default (state = menuItemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_MENU_ITEM':
      return [...state, action.menuItem];
    case 'REMOVE_MENU_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_MENU_ITEM':
      return state.map(menuItem => {
        if (menuItem.id === action.id) {
          return {
            ...menuItem,
            ...action.updates
          };
        } else {
          return menuItem;
        }
      });
    default:
      return state;
  }
};
