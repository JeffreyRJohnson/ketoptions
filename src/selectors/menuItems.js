// Get visible menuItems

export default (menuItems, { text }) => {
  return menuItems.filter(menuItem => {
    const textMatch = menuItem.restaurant_name
      .toLowerCase()
      .includes(text.toLowerCase());

    return textMatch;
  });
};
