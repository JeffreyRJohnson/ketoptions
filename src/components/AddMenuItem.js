import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { startAddMenuItem } from '../actions/menuItems';

export class AddMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.startAddMenuItem(menuItem);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Menu Item</h1>
        <MenuItemForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddMenuItem: menuItem => dispatch(startAddMenuItem(menuItem))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddMenuItem);
