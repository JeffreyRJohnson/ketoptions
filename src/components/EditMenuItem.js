import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { editMenuItem, removeMenuItem } from '../actions/menuItems';

export class EditMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.editMenuItem(this.props.menuItem.id, menuItem);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeMenuItem({ id: this.props.menuItem.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <MenuItemForm menuItem={this.props.menuItem} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  menuItem: state.menuItems.find(
    menuItem => menuItem.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  editExpense: (id, menuItem) => dispatch(editMenuItem(id, menuItem)),
  removeMenuItem: data => dispatch(removeMenuItem(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditMenuItem);
