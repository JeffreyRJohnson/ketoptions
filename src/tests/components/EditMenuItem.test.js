import React from 'react';
import { shallow } from 'enzyme';
import { EditMenuItem } from '../../components/EditMenuItem';
import menuItems from '../fixtures/menuItems';

let editMenuItem, removeMenuItem, history, wrapper;

beforeEach(() => {
  editMenuItem = jest.fn();
  removeMenuItem = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditMenuItem
      editMenuItem={editMenuItem}
      removeMenuItem={removeMenuItem}
      history={history}
      menuItem={menuItems[2]}
    />
  );
});

test('should render EditMenuItem', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle editMenuItem', () => {
  wrapper.find('MenuItemForm').prop('onSubmit')(menuItems[2]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(editMenuItem).toHaveBeenLastCalledWith(menuItems[2].id, menuItems[2]);
});

test('should handle removeMenuItem', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeMenuItem).toHaveBeenLastCalledWith({
    id: menuItems[2].id
  });
});
