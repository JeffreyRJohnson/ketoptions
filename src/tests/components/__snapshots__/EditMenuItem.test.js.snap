// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditMenuItem 1`] = `
<div>
  <MenuItemForm
    menuItem={
      Object {
        "calories": 600,
        "carbs": 80,
        "fat": 85,
        "id": "3",
        "menu_item": "chilli - large",
        "protein": 50,
        "restaurant_name": "Burger King",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
